src/utils/memory_tracker.hpp

```diff
@@ -49,6 +49,7 @@ class MemoryTracker final {
 
   void Alloc(int64_t size);
   void Free(int64_t size);
+  void DoCheck();
```

Comment:
But don't we do this check already in `MemoryTracker::Alloc` whenever `utils::MemoryTracker::OutOfMemoryExceptionEnabler` exists?

Reply:
That would require additional logic and passing some flags or something when doing tracking from global_memory_tracker, so I wouldn't go there unless it is necessary.